service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -448,7 +450,21 @@ public String cloneWorkspace(
         String.format("Clone workspace: name: '%s' id: '%s'  ", workspaceName, workspaceUuid);
 
     // Create the destination workspace synchronously first.
-    // TODO(PF-1871) copy policy on workspace clone
+    if (features.isTpsEnabled()) {
+      // New workspaces will always be created with empty policies, but some workspaces predate
+      // policy and so will not have associated PAOs.
+      Optional<ApiTpsPaoGetResult> workspacePao =
+          tpsApiDispatch.getPaoIfExists(
+              new BearerToken(userRequest.getRequiredToken()), sourceWorkspace.getWorkspaceId());
+
+      if (workspacePao.isPresent()) {
+        tpsApiDispatch.clonePao(
```

Comment:
I think cloning the PAO needs to be a step in `CloneGcpWorkspaceFlight` rather than here own its own. 
When it's outside a flight like this it won't be undone, so if the flight in the `createWorkspace` call below fails, we will undo all the workspace creation work and return an error but won't undo the policy clone, and would have a policy object for a workspace which doesn't exist

Reply:
I switched this to a step in the Clone flight. It still feels a little funny because the Create flight adds a default policy that we need to clean later. I think that's why I had it as a sync pre workspace step before.

Reply:
ah, good point. The whole pattern of creating the destination workspace in one flight and then cloning into it in another feels pretty weird, but I think that's required in order for the "clone workspace" job to live in the destination workspace and definitely gets outside the scope of this PR